src/api/configure.ts

```diff
@@ -0,0 +1,34 @@
+import {Request, Response} from 'express';
+
+import {
+    CallResponseHandler,
+    newFormCallResponse,
+    newOKCallResponseWithMarkdown,
+} from '../utils/call-responses';
+import {AppCallResponse} from '../types';
+import {googleClientConfigForm, googleClientConfigFormSubmit} from '../forms/configure-google-client';
+import {showMessageToMattermost} from '../utils/utils';
+
+export const configureGoogleClient: CallResponseHandler = async (req: Request, res: Response) => {
```

Comment:
This was handled in each form (`googleClientConfigForm` and  `googleClientConfigFormSubmit`)
```
    const actingUser: AppActingUser = call.context.acting_user as AppActingUser;
    if (!isUserSystemAdmin(<AppActingUser>actingUser)) {
        throwException(ExceptionType.TEXT_ERROR, i18nObj.__('configure-binding.error.system-admin'));
    }
```

Reply:
The code you provide here is not in `googleClientConfigForm` or `googleClientConfigFormSubmit`, so it's not being handled.

I'm thinking the logic around what endpoints are accessible to specific users should be handled on the API layer. The code is much easier to read this way. The API then acts as a gate to the functions that actually handle the specific request.

Reply:
We can handle this for multiple endpoints using express middleware

Reply:
Since there are no automated tests that are checking for these access control restrictions, the  checks should be made easily discoverable and reviewable. Check out this PR for an example https://github.com/mattermost/mattermost-app-zendesk/pull/98

Reply:
Resolved in [dae163f](https://github.com/liz-segura98/mattermost-app-google-drive/commit/dae163f15b39d8265784fa0a9b8f27d9af5672cd)